File: todo-back/infrastructure/repository/session.go
package infrastructure

import (
	"todo-back/domain/model"
	"todo-back/domain/repository"
	"todo-back/infrastructure/cloudflare"
)

type session struct {
	workersKVClient cloudflare.WorkersKVClient
}

func NewSession(cloudflareApiToken string, workersKVClient cloudflare.WorkersKVClient) repository.Session {
	return &session{
		workersKVClient: workersKVClient,
	}
}

func (s *session) Insert(session model.Session, userID model.UserID) error {

	panic("todo")
}
func (s *session) Delete(session model.Session) error {

	panic("todo")
}
func (s *session) Get(sessionID string) (model.UserID, error) {

	panic("todo")
}

File: todoapp/lib/infrastructure/repository/session.dart
import 'package:todoapp/domain/model/user_auth.dart';
import 'package:todoapp/domain/repository/session.dart' as $repository;

class SessionRepository implements $repository.SessionRepository {
  @override
  Future<void> delete() {
    // TODO: implement delete
    throw UnimplementedError();
  }

  @override
  Future<Session> get() {
    // TODO: implement get
    throw UnimplementedError();
  }

  @override
  Future<void> set(Session session) {
    // TODO: implement set
    throw UnimplementedError();
  }
}
